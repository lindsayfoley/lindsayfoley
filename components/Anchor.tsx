import React from "react";

export interface IAnchorProps {
  link: string;
  icon?: string;
  cta?: string;
  title?: string;
  external?: boolean;
}

const Anchor: React.FunctionComponent<IAnchorProps> = ({
  link,
  icon,
  cta,
  title,
  external
}) => external ? (
  <>
    <a target="_blank" rel="noreferrer" href={link} title={title}>
      {icon && <i className={icon}></i>}{cta}
    </a>
  </>
) : (
      <>
        <a href={link} title={title}>
          {icon && <i className={icon}></i>}{cta}
        </a>
      </>
    )


export default Anchor;