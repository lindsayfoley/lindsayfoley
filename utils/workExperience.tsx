import React from "react";

export interface ICompanyDetails {
  id: string;
  companyName: string;
  description: JSX.Element;
  cta?: string;
  link: string;
}

const workExperience: ICompanyDetails[] = [
  {
    id: "travel-blog",
    companyName: "Come Fly With Lindsay",
    description: (
      <>
        <p>My travel blog which I launched in November 2016.</p>
        <p>I designed and coded it using <span itemProp="keywords"> PHP, Markdown, HTML, CSS, jQuery &amp; Foundation.</span></p>
      </>
    ),
    link: "http://comeflywithlindsay.com",
  },
  {
    id: "disney",
    companyName: "Disney Holidays",
    description: (
      <p>I co-developed this site making it adaptive &amp; responsive using <span itemProp="keywords"> Foundation &amp; RESS principles</span> resulting in a better UX.</p>
    ),
    link: "http://www.disneyholidays.co.uk",
  },
  {
    id: "jia",
    companyName: "Jesus In Action",
    description: (
      <>
        <p>This website underwent a complete refresh in order to modernise it.</p>
        <p>It's written in <span itemProp="keywords">HTML, CSS, jQuery &amp; Foundation</span>.</p>
      </>
    ),
    link: "http://jesusinaction.org.uk",
  },
  {
    id: "perfume",
    companyName: "The Perfume Mistress",
    description: (
      <p>This one is on Squarespace as my client needed a transactional site, it has bespoke <span itemProp="keywords">CSS &amp; JavaScript</span> to customise the Hayden template.</p>
    ),
    link: "https://www.theperfumemistress.com",
  },
  {
    id: "dashwood",
    companyName: "Hayes Dashwood",
    description: (
      <p>I redesigned and coded this website using <span itemProp="keywords"> HTML, CSS, jQuery &amp; the Google Maps API</span>.</p>
    ),
    link: "http://www.hayesdashwood.org.uk",
  },
  {
    id: "selfridges",
    companyName: "Selfridges",
    description: (
      <>
        <p>I worked on Selfridges' 2014 site redesign.</p>
        <p>I migrated and <span itemProp="keywords"> developed new responsive content &amp; created HTML emails</span>.</p>
      </>
    ),
    cta: "View past work ›",
    link: "https://web.archive.org/web/20150707135538/http://www.selfridges.com/content/article/paddington",
  },
  {
    id: "hopelands",
    companyName: "Hopelands Cattery",
    description: (
      <p>I designed and developed this site using <span itemProp="keywords"> HTML, CSS, jQuery, Slick Slider &amp; the Google Maps API</span>.</p>
    ),
    link: "https://web.archive.org/web/20190625163932/http://hopelandscattery.com/",
  },
  {
    id: "whitestuff",
    companyName: "White Stuff",
    description: (
      <p>My first ever dev role! I created web content and weekly <span itemProp="keywords"> HTML emails using XHTML4, CSS &amp; jQuery</span>.</p>
    ),
    cta: "View past work ›",
    link: "http://web.archive.org/web/20140511040437/http://www.whitestuff.com/jobs",
  },
  {
    id: "ppp",
    companyName: "Paw Print Productions",
    description: (
      <p>I co-developed and designed this responsive site many years ago (2011) using <span itemProp="keywords">HTML, CSS &amp; a bit of jQuery</span>.</p>
    ),
    link: "https://web.archive.org/web/20160115162243/http://paw-print-productions.co.uk",
  },
];

export default workExperience;