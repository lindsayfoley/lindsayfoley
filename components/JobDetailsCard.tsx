import React from "react";
import Anchor from "./Anchor";
import { ICompanyDetails } from "../utils/workExperience";
import "../public/styles/_flip-card.scss";

export interface IJobDetailsCardProps {
  company: ICompanyDetails;
  className?: string;
  handleClick: (id: string) => void;
  summaryIsVisible: boolean;
}

const JobDetailsCard: React.FunctionComponent<IJobDetailsCardProps> = ({ 
  company, className, handleClick, summaryIsVisible 
}) => {

  if (!company) {
    return null;
  }

  const defaultClassName = 'flip-card';
  const classNames = className !== "" && `${className} ${defaultClassName}`;

  const handleOnKeyDown = (event: React.KeyboardEvent<HTMLElement>) => {
    if (event.key === " " || event.key === "Enter" || event.key === "Spacebar") {
      handleClick(company.id);
    }
  }

  return (
    <article
      id={company.id}
      className={classNames || defaultClassName}
      onClick={() => handleClick(company.id)}
      onKeyDown={handleOnKeyDown}
      aria-controls={company.id}
      role="button" 
      tabIndex={0}
      aria-pressed={summaryIsVisible}
    >
      <div className="wrapper" itemScope itemType="http://schema.org/CreativeWork">
        <div className="heading">
          <h3 itemProp="name">{company.companyName}</h3>
          <span>Details &rsaquo;</span>
        </div>
        <div 
          className="description" 
          itemProp="description" 
          aria-hidden={!summaryIsVisible}
          aria-labelledby={company.id}
        >
          {company.description}
          <hr />
          {company.cta ? (
            <Anchor link={company.link} cta={company.cta} />
          ) : (
            <Anchor link={company.link} cta="Visit Site &rsaquo;" />
          )}
        </div>
      </div>
    </article>
  );
}

export default JobDetailsCard;
