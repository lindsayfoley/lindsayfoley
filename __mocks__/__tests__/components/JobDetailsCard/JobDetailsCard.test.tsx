import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { ICompanyDetails } from '../../../../utils/workExperience';
import JobDetailsCard, { IJobDetailsCardProps } from '../../../../components/JobDetailsCard';
import Anchor from '../../../../components/Anchor';

const companyDetailsDefaultProps: ICompanyDetails = {
    id: 'scampi',
    companyName: 'Scampi The Cat',
    description: <p>Bring your cat to work at Scampi The Cat!</p>,
    link: 'scampithecat.com'
  }

const defaultProps: IJobDetailsCardProps = {
    company: companyDetailsDefaultProps,
    className: 'testClass',
    handleClick: jest.fn(),
    summaryIsVisible: true
}

const setup = (props = {}) => {
    const setupProps: IJobDetailsCardProps = { ...defaultProps, ...props }
    return shallow(<JobDetailsCard {...setupProps} />);
};

describe('JobDetailsCard', () => {

    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = setup();
    });

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('renders all required card details', () => {
        expect(wrapper.find('article')).toHaveLength(1);
        expect(wrapper.find('h3').text()).toContain(defaultProps.company.companyName);
        expect(wrapper.find('.description')).toHaveLength(1);
        expect(wrapper.find(Anchor).props()).toHaveProperty('link', defaultProps.company.link);
    });

    it('doesn\'t render the class to toggle visibility of descriptions by default', () => {
        expect(wrapper.find('.show-details')).toHaveLength(0);
    });

    it('fires an onClick event when job card is clicked', () => {
        wrapper.find('.flip-card').simulate('click');
        expect(defaultProps.handleClick).toBeCalledTimes(1);
    });
});
