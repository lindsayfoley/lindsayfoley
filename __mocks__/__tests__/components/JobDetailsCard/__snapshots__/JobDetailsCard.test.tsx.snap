// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JobDetailsCard renders correctly 1`] = `
<article
  aria-controls="scampi"
  aria-pressed={true}
  className="testClass flip-card"
  id="scampi"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
>
  <div
    className="wrapper"
    itemScope={true}
    itemType="http://schema.org/CreativeWork"
  >
    <div
      className="heading"
    >
      <h3
        itemProp="name"
      >
        Scampi The Cat
      </h3>
      <span>
        Details ›
      </span>
    </div>
    <div
      aria-hidden={false}
      aria-labelledby="scampi"
      className="description"
      itemProp="description"
    >
      <p>
        Bring your cat to work at Scampi The Cat!
      </p>
      <hr />
      <Anchor
        cta="Visit Site ›"
        link="scampithecat.com"
      />
    </div>
  </div>
</article>
`;
